### PR TITLE
magento-engcom/community-portal#174 Display Custom Error Messages for User Actions

### PR DESCRIPTION
**Pull request by XiyaMagento**
magento-engcom/community-portal#174 Display Custom Error Messages for User Actions

- [x] Remove GitID l output
- [x] Display custom error messages for actions including `pledge`, `likeProject`, `bookmarkProject`, `createProject` and `editProject` when rejected
- [x] Remove error catch block within dispatch so it can be caught in the outer components
- [x] Change fetch to axios because fetch resolves non-200 reponses
- [ ] Fix frontend test

### CODE REVIEW

File: frontend/src/actions/index.ts [hunk 1]
```diff
@@ -86,6 +86,7 @@ export const addProject = (project: {}) => {
   return (dispatch: Dispatch) => {
     return saveProject(projectBody)
       .then(() => {
+        console.log('then called');
```
Comment (vrann) [inline frontend/src/actions/index.ts]: please remove debug

File: frontend/src/components/ProjectCard.tsx [hunk 2]
```diff
@@ -270,7 +277,7 @@ export class ProjectCard extends React.Component<CardProps & DispatchProps, Card
           this.toggleLike();
         })
         .catch((err: Error) => {
-          console.error(err);
+          this.onFailure(new Error('Oops, something went wrong while liking this project'));
```
Comment (vrann) [inline frontend/src/components/ProjectCard.tsx]: please remove 'Ooops'

Comment (vrann) [reply frontend/src/components/ProjectCard.tsx]: here and everywhere

Comment (XiyaMagento) [reply frontend/src/components/ProjectCard.tsx]: 'Oops' have been removed from the error messages

**Pull request merged**